URL-tools/plugins/urltools/handlers/osu/osu.py

```diff
@@ -0,0 +1,409 @@
+# coding=utf-8
+import re
+import locale
+
+from twisted.internet.defer import inlineCallbacks, returnValue
+from txrequests import Session
+
+from plugins.urls.handlers.handler import URLHandler
+from plugins.urltools.exceptions import ApiKeyMissing
+from plugins.urltools.handlers.osu.mods import get_mods
+from utils.misc import str_to_regex_flags
+
+__author__ = 'Gareth Coles'
+
+URL_BASE = "https://osu.ppy.sh/api"
+
+URL_BEATMAPS = URL_BASE + "/get_beatmaps"
+URL_MATCH = URL_BASE + "/get_match"
+URL_REPLAY = URL_BASE + "/get_replay"
+URL_SCORES = URL_BASE + "/get_scores"
+URL_USER = URL_BASE + "/get_user"
+URL_USER_BEST = URL_BASE + "/get_user_best"
+URL_USER_RECENT = URL_BASE + "/get_user_recent"
+
+# Attempt to guess the locale
+locale.setlocale(locale.LC_ALL, "")
```

Comment:
This should really be called once on bot startup then never again. Also, it's called in main urltools init file anyway.